src/micro-emitter.ts

```diff
@@ -14,44 +18,53 @@ export interface IEventEmitter {
      * @param that - (optional) The context with which to call onFunction (useful also for unsubscribing only a instance)
      */
 
-    on(eventType: string, onFunction: Callback, that?: any): this;
+    on<Event extends keyof Events>(
+        eventType: Event,
+        onFunction: Events[Event],
+        that?: any,
+    ): any;

```

Comment:
Why can't you return this any more?

Reply:
I change it so that we are not forced to implement chaining. Was thinking about our case in SaxoTrader where we don't want it in some cases. TS is able to infer the return type when a particular implementation is used. The only drawback is when someone expects just an interface, in such a case return type will by any. I tried to make it generic but we can't use this in a static context. Also [tried this](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcCiA3YA7GAVZYAZzgF44AlYAY2gBMAeImKAS2wHMAaOACgDpBAQygciALjhDsiANoBdAJRkAfHAwRWdFQG4AsAChDAemNxQkWHHYxgUAGZDqqAJKYcMNAFtWMW1AZ3XBJQW2w6EiD8QiIeCjIEAAtWIjUAbxMzC2h4GztHZzg3LFxvX39AkpgQkDCI9CqCFFjKBOKPMr87So9U9LhDbKs8hydXKM6KqJq6yMaYuLaJny6A6Z5pRBV+wzg9uAhsHtxzWpx6gGtgRAh7Bt6VXl39l+B5lEkormeXvcOAMQArthqDBWIdPlUiLIovJvgZfvsYIkhDAAPySTbwl6KSQUfQGAC+hkycDwUEQCAgcGoKPYHAO2DgXmAyIgdDgiTswEM9mBoPBTNszF4IEk7VKKwqaWFMEkvCEkmYbE4ylIag0WkJKmUGQR+xA-EOvAA5LKTTxeGr0oTFD84EbsKb7BBXRa+Na4GlbXBTHAAOq+RKtcgSzxS7rrKQybZkgDK1hIQgARgAbVAwal0Vl2HzYVBCOBgKAQFBQcxVBCERmpymZpIpODprCp+2O03JkTuq2qL22gnEowGIA) but I had to use `any` since otherwise, we have a circular default if we make `R = IEventEmitter<Events>`